robogen/src/main.rs

```diff
@@ -0,0 +1,84 @@
+use std::io::Write;
+
+
+#[derive(Debug, Copy, Clone)]
+struct PixelState {
+    hvis: bool,
+    hsync: bool,
+    vvis: bool,
+    vsync: bool,
+    frame_vis: bool,
+    end: bool
+}
+
+impl Default for PixelState {
+    fn default() -> Self {
+        PixelState { hvis: false, hsync: true, vvis: false, vsync: true, frame_vis: false, end: false }
+    }
+}
+
+impl From<PixelState> for u8 {
+    fn from(p: PixelState) -> u8 {
+        p.hvis as u8 | (p.hsync as u8) << 1 | (p.vvis as u8) << 2 | (p.vsync as u8) << 3 | (p.frame_vis as u8) << 4 | (p.end as u8) << 5
```

Comment:
maybe we should merge the schematic, so we can keep this part & that changing together? It doesn't have hvis or vvis broken out as individual signals (which is fine, there's 4 unallocated bits on the drawing), but I arbitrarily chose to put HSync, VSync, and Vis (here, frame_vis) on bits 1, 2, and 3 respectively, and /RESET in bit 8 (which would be equivalent to !end, here)

Reply:
Yeah that makes sense. Wonder if in the Future we could somehow ~generate~ the schematic, or paramterize it somehow based on the code here? Or parameterize the code and schematic based on a single input? But yes, good idea, I wasn't even thinking about that. We could totally make the end bit bit 7, and switch the others around. I think of the horizontal things being grouped and the vertical things being grouped, but it doesn't matter much, esp if the bitmask stuff makes a nice type either way :)